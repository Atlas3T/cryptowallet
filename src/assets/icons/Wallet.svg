<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
