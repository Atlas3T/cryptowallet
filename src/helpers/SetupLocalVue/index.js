import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
