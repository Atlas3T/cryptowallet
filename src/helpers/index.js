import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
