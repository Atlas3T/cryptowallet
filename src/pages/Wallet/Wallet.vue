<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
