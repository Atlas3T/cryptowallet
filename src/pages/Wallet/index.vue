<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
