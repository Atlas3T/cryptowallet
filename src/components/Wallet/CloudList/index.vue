<template>
  <div>
    <div v-if="wallets.length === 0">
      <q-btn
        icon="add_circle_outline"
        label="add"
        color="primary"
        size="xl"
        class="large-cloud-btn"
        @click.prevent="openWalletsModal"
      />
    </div>
    <q-scroll-area
      v-if="wallets.length > 0"
      ref="scrollArea"
      class="scroll-area extended cloud-scroll"
    >
      <div class="scroll-offset" />

      <!-- <CloudListItem
        v-for="wallet in wallets"
        :key="wallet.displayName"
        :wallet="wallet"
        :currency="selectedCurrency"
      /> -->
    </q-scroll-area>
  </div>
</template>

<script>
// import CloudListItem from '@/components/Wallet/CloudListItem';
import { mapState } from 'vuex';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';

export default {
  name: 'CloudList',
  components: {
    // CloudListItem,
  },

  data() {
    return {
      scrollPosition: 0,
    };
  },

  computed: {
    ...mapState({
      authenticatedAccount: (state) => { return state.settings.authenticatedAccount; },
    }),

    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    showTestnets() {
      return this.$store.getters['entities/account/find'](this.authenticatedAccount).showTestnets;
    },

    testnets() {
      const coins = Coin.query()
        .where('testnet', true).get();
      return coins.map(({ network }) => { return network; });
    },

    wallets() {
      const wallets = Wallet.query()
        .where('account_id', this.authenticatedAccount)
        .where('imported', true).get();
      if (!this.showTestnets) {
        return wallets.filter(({ network }) => {
          return !this.testnets.includes(network);
        });
      }
      return wallets;
    },
  },
  activated() {
    if (document.querySelectorAll('.cloud-scroll .scroll')[0]) {
      document.querySelectorAll('.cloud-scroll .scroll')[0].scrollTop = this.scrollPosition;
    }
  },

  methods: {
    openWalletsModal() {
      this.$store.dispatch('modals/setAddWalletModalOpened', true);
    },

    touchStart(event) {
      this.touchStartY = event.touches[0].clientY;
    },

    touchMove(event) {
      if (event.touches[0].clientY <= this.touchStartY) {
        event.stopPropagation();
      }

      if (this.wallets.length === 0 || !this.$refs.scrollArea) { return false; }
      if (this.$refs.scrollArea.$el.childNodes[0].scrollTop !== 0) {
        event.stopPropagation();
      }

      return true;
    },

    scrolled(data) {
      this.scrollPosition = data.position;
      const pixels = 100;
      if (data.position > pixels && data.direction === 'down') {
        this.$root.$emit('isHomeBalanceVisible', false);
      }
      if (data.position <= pixels && data.direction === 'up') {
        this.$root.$emit('isHomeBalanceVisible', true);
      }
    },
  },
};
</script>

<style>
.scroll-area {
  position: absolute;
  height: 100%;
  width: 100%;
  left: 0;
  top: 0;
}

.large-cloud-btn {
  width: 100%;
  height: 3em;
  border-radius: 0.4rem;
  font-size: 3em;
  padding: 0em;
  color: #1e3c57;
  margin-top: 5rem;

}

.large-cloud-btn i {
  color: #1e3c57;
}

.large-cloud-btn .q-btn__content{
  color: #1e3c57;
}

.scroll-offset {
  height: 20rem;
}
</style>
