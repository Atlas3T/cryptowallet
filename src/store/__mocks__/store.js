import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
