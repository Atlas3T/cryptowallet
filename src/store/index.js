/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
