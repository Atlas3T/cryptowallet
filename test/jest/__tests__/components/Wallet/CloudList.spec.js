/* eslint-disable no-magic-numbers */
import CloudList from '@/components/Wallet/CloudList';
import { mount, createWrapper } from '@vue/test-utils';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Wallet from '@/store/wallet/entities/wallet';

describe('CloudListItem.vue', () => {
  let storeMocks;
  let wrapper;
  let router;

  const mockWallets = {
    data: [{
      id: 1,
      account_id: 1,
      name: 'Catalyst',
      displayName: 'Catalyst',
      parentName: 'Ethereum',
      symbol: 'CAT',
      network: 'ETHEREUM_ROPSTEN',
      sdk: 'ERC20',
      enabled: false,
      imported: true,
    },
    {
      id: 2,
      account_id: 1,
      name: 'Bitcoin',
      displayName: 'Bitcoin',
      symbol: 'BTC',
      network: 'BITCOIN',
      sdk: 'Bitcoin',
      enabled: true,
      imported: true,
    },
    {
      id: 5,
      account_id: 1,
      name: 'Ethereum',
      displayName: 'Ethereum',
      sdk: 'Ethereum',
      symbol: 'ETH',
      enabled: true,
      imported: true,
    },
    ],
  };

  function wrapperInit(options) {
    return mount(CloudList, options);
  }

  function storeInit(custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: '/wallet' });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      store: storeMocks.store,
      stubs: { CloudListItem: true },
    });
  }

  it('renders and matches snapshot', () => {
    storeInit();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays an add wallet button if no wallets are enabled', () => {
    storeInit();
    expect(wrapper.find('button').text()).toMatch('add_circle_outline');
  });

  it('calls setAddWalletModalOpened action on add walet button click', (done) => {
    storeInit();
    wrapper.find('button').trigger('click');
    setTimeout(() => {
      expect(storeMocks.actions.setAddWalletModalOpened.mock.calls[0][1]).toEqual(true);
      done();
    }, 0);
  });

  it('displays a cloud list item for each enabled wallet', (done) => {
    storeInit();
    Wallet.$insert(mockWallets);
    setTimeout(() => {
      wrapper.vm.$options.activated[0]();
      expect(wrapper.findAll('cloudlistitem-stub').length).toEqual(3);
      done();
    }, 0);
  });

  it('calls the touchStart() method on touchstart event', () => {
    storeInit();
    wrapper.find('.scroll-start').trigger('touchstart', { touches: [{ clientY: 399 }] });
    expect(wrapper.vm.touchStartY).toBe(399);
  });

  it('calls the touchMove() method on touchMove event', () => {
    storeInit();
    Wallet.$insert(mockWallets);
    const mockStopPropagation = jest.fn();
    wrapper.find('.scroll-start').trigger('touchstart', { touches: [{ clientY: 399 }] });
    wrapper.find('.scroll-start').trigger('touchmove', { touches: [{ clientY: 400 }] });
    expect(wrapper.vm.touchStartY).toBe(399);
    wrapper.vm.$refs.scrollArea.$el.childNodes[0].scrollTop = 10;
    wrapper.find('.scroll-start').trigger('touchstart', { touches: [{ clientY: 399 }] });
    wrapper.find('.scroll-start').trigger('touchmove', { stopPropagation: mockStopPropagation, touches: [{ clientY: 399 }] });
    expect(mockStopPropagation).toHaveBeenCalledTimes(2);
  });

  it('fires an event to hide and show the balance when the user scrolls', (done) => {
    storeInit();
    const rootWrapper = createWrapper(wrapper.vm.$root);
    Wallet.$insert(mockWallets);
    wrapper.vm.scrolled({ position: 101, direction: 'down' });
    setTimeout(() => {
      expect(rootWrapper.emitted('isHomeBalanceVisible')[0]).toEqual([false]);
      done();
    }, 0);
    wrapper.vm.scrolled({ position: 99, direction: 'up' });
    setTimeout(() => {
      expect(rootWrapper.emitted('isHomeBalanceVisible')[1]).toEqual([true]);

      done();
    }, 0);
  });

  it('activated lifecycle hook resets scroll position', (done) => {
    const mockQuerySelector = jest.fn().mockImplementation(() => {
      return [{ scrollTop: 10 }];
    });
    Object.defineProperty(global.document, 'querySelectorAll', {
      value: mockQuerySelector,
    });
    storeInit();
    wrapper.vm.$options.activated[0]();
    setTimeout(() => {
      expect(mockQuerySelector).toHaveBeenCalledTimes(2);
      done();
    }, 0);
  });
});
