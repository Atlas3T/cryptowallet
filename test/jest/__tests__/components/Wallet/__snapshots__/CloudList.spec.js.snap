// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudListItem.vue renders and matches snapshot 1`] = `
<div
  class="scroll-start"
>
  <div>
    <button
      class="q-btn inline q-btn-item non-selectable large-cloud-btn q-btn--standard q-btn--rectangle bg-primary text-white q-focusable q-hoverable"
      style="font-size: 24px;"
      tabindex="0"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <i
          aria-hidden="true"
          class="material-icons q-icon on-left"
        >
          add_circle_outline
        </i>
        <div>
          add
        </div>
      </div>
    </button>
  </div>
   
  <!---->
</div>
`;
